src/main/java/edu/umsystem/RecursiveQuickSort.java

```diff
@@ -0,0 +1,62 @@
+package edu.umsystem;
+
+public class RecursiveQuickSort implements SortingAlgorithm {
+    SortShow arrayDisplay;
+
+    RecursiveQuickSort(SortShow arrayDisplay) {
+        this.arrayDisplay = arrayDisplay;
+    }
+
+    @Override
+    public String getName() {
+        return "Recursive Quick Sort";
+    }
+
+    @Override
+    public void sort() {
+        quickSort(arrayDisplay.lines_lengths, 0, arrayDisplay.total_number_of_lines - 1);
+    }
+
+    private void quickSort(int[] a, int first, int last) {
+        if (first < last) {
+            // Split array into three partitions: [first, pivot - 1], [pivot] (sorted), and [pivot + 1, last]
+            int pivot = partition(a, first, last);
+
+            // Sort first and third partitions
+            quickSort(a, first, pivot - 1);
+            quickSort(a, pivot + 1, last);
+        }
+    }
+
+    private int partition(int[] a, int first, int last) {
+        int left = first;
+        int right = last - 1;
+        int pivot = a[last];
+
+        while (left <= right) {
```

Comment:
sorry this is probably just my misunderstanding but what exactly are the invariants for the left and right partitions? to me it seems like the left side contains elements <= pivot while the right contains elements >= pivot with elements == pivot sort of split between the two.